seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py

```diff
@@ -0,0 +1,196 @@
+                    'tokens'  : datasets.Sequence(datasets.Value("string")),
+                    'ner_tags': datasets.Sequence(
+                                    datasets.ClassLabel(names=["O", "B-WIS", "I-WIS", "B-LOC", "I-LOC", "B-FAS", "I-FAS"])
+                                ),
+                }
+            )
+        elif self.config.schema == "seacrowd_seq_label":
+            features = schemas.seq_label.features(["O", "B-WIS", "I-WIS", "B-LOC", "I-LOC", "B-FAS", "I-FAS"])
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
+        path = dl_manager.download_and_extract(urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": path,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        tokens = []
+        ner_tags = []
+        counter = 0
+        with open(filepath, encoding="utf-8") as file:
+            for line in file:
+                if line.strip() == "":
+                    if self.config.schema == "source":
+                        yield(
+                            counter,
+                            {
+                                'tokens'  : tokens,
+                                'ner_tags': ner_tags,
+                            }
+                        )
+                        counter += 1
+                        tokens = []
+                        ner_tags = []
+                    elif self.config.schema == "seacrowd_seq_label":
+                        yield(
+                            counter,
+                            {
+                                'id'    : counter,
+                                'tokens': tokens,
+                                'labels': ner_tags,
+                            }
+                        )
+                        counter += 1
+                        tokens = []
+                        ner_tags = []
+                elif len(line.split('\t')) == 2:
+                    token, ner_tag = line.split('\t')
+                    tokens.append(token.strip())
+                    if ner_tag not in ["O", "B-WIS", "I-WIS", "B-LOC", "I-LOC", "B-FAS", "I-FAS"]:
+                        if ner_tag[0] in ["B", "I"]:
+                            if any(tag in ner_tag for tag in ["WIS", "LOC", "FAS"]):
+                                if '_' in ner_tag:
+                                    ner_tag = '-'.join(ner_tag.split('_'))
```

Comment:
Can you add a brief comment on what these lines do?

Reply:
Comment added!